moveos/moveos-stdlib/moveos-stdlib/sources/object.move

```diff
@@ -50,6 +50,10 @@ module moveos_std::object {
         &self.value
     }
 
+    public fun public_borrow<T: key + store>(self: &Object<T>): &T {
```

Comment:
How about we remove the `private_generics` from `borrow` and `borrow_mut`?  Because `Context::borrow_object` and `Context::borrow_object_mut` has the `private_generics`.

@stevenlaw123 @pause125 @baichuan3

Reply:
And should we support borrowing Object<T: key+store> from Context without `private_generics` ?

Reply:
Wait a minute, I feel there is a problem here. If we let go of Borrow, this means that T can be obtained by any module at will, which may cause security problems.

So I think it’s better to delete them and add them if necessary.

Reply:
> And should we support borrowing Object<T: key+store> from Context without `private_generics` ?

Maybe this shouldn't be the case, it would mean that other modules could borrow arbitrary Object<T>
This may cause security issues

Reply:
> How about we remove the `private_generics` from `borrow` and `borrow_mut`? Because `Context::borrow_object` and `Context::borrow_object_mut` has the `private_generics`.
> 
> @stevenlaw123 @pause125 @baichuan3
I think it's reasonable
Because there is no way to get `&Object<T>` or `Object<T>` except context and current module.

Reply:
> borrowing Object<T: key+store>

The reference of Object can only be obtained through context. I think it can be removed and there is no security issue.

Reply:
In my understanding, borrow Object<T: key> and borrow Object<T: key + store> satisfy two types of scenarios respectively?
1. If Object<T: key> is only a key capability, it can only be accessed through context::borrow_object (with private_generics(T) restriction) or object:borrow (need to pass Object<T> reference);
2. If Object<T: key+store> also has store capability, in addition to option 1, it can also be called through context::public_borrow_object* (there is no private_generics(T) restriction, and there is no need to pass the Object<T> reference).